l3kernel/l3str.dtx

```diff
@@ -1306,7 +1348,7 @@
   { #1 { ~ { ~ } \@@_map_function:w } }
 \cs_new:Npn \@@_map_function:Nn #1#2
   {
-    \if_meaning:w \q_recursion_tail #2
+    \if_meaning:w \q_@@_recursion_tail #2
       \exp_after:wN \str_map_break:
     \fi:
```

Comment:
I'm not sure why here we (most likely I) chose not to use the `if_recursion_tail_break` function.

Reply:
My guess was performance... Comparing both versions I get:
```none
1.78e-6 seconds (7.83 ops)
6.04e-6 seconds (26.7 ops)
```
for the `\if_meaning:w` and `\quark_if_recursion_tail_break:nN` versions,
respectively for `\def\test#1{}` and `\str_map_function:nN{a~b}\test`.
For an input string 20 times as long, the run time is about 10 times
longer for both cases.